adafruit_portalbase/network.py

```diff
@@ -325,36 +340,45 @@ def connect(self, max_attempts=10):
                              failing or use None to disable. Defaults to 10.
 
         """
-        self._wifi.neo_status(STATUS_CONNECTING)
-        attempt = 1
-        while not self._wifi.is_connected:
-            # secrets dictionary must contain 'ssid' and 'password' at a minimum
-            print("Connecting to AP", self._secrets["ssid"])
-            if (
-                self._secrets["ssid"] == "CHANGE ME"
-                or self._secrets["password"] == "CHANGE ME"
-            ):
-                change_me = "\n" + "*" * 45
-                change_me += "\nPlease update the 'secrets.py' file on your\n"
-                change_me += "CIRCUITPY drive to include your local WiFi\n"
-                change_me += "access point SSID name in 'ssid' and SSID\n"
-                change_me += "password in 'password'. Then save to reload!\n"
-                change_me += "*" * 45
-                raise OSError(change_me)
-            self._wifi.neo_status(STATUS_NO_CONNECTION)  # red = not connected
-            try:
-                self._wifi.connect(self._secrets["ssid"], self._secrets["password"])
-                self.requests = self._wifi.requests
-            except ConnectionError as error:
-                if max_attempts is not None and attempt >= max_attempts:
-                    raise OSError(
-                        "Maximum number of attempts reached when trying to connect to WiFi"
-                    ) from error
-                print("Could not connect to internet", error)
-                print("Retrying in 3 seconds...")
-                attempt += 1
-                time.sleep(3)
-            gc.collect()
+        for secret_entry in self._secrets_network:
+
+            self._wifi.neo_status(STATUS_CONNECTING)
+            attempt = 1
+
+            while not self._wifi.is_connected:
+                # secrets dictionary must contain 'ssid' and 'password' at a minimum
+                print("Connecting to AP", secret_entry["ssid"])
+                if (
+                    secret_entry["ssid"] == "CHANGE ME"
+                    or secret_entry["password"] == "CHANGE ME"
+                ):
+                    change_me = "\n" + "*" * 45
+                    change_me += "\nPlease update the 'secrets.py' file on your\n"
+                    change_me += "CIRCUITPY drive to include your local WiFi\n"
+                    change_me += "access point SSID name in 'ssid' and SSID\n"
+                    change_me += "password in 'password'. Then save to reload!\n"
+                    change_me += "*" * 45
+                    raise OSError(change_me)
+                self._wifi.neo_status(STATUS_NO_CONNECTION)  # red = not connected
+                try:
+                    self._wifi.connect(secret_entry["ssid"], secret_entry["password"])
+                    self.requests = self._wifi.requests
+                    break
+                except ConnectionError as error:
```

Comment:
In order to get the multiple networks to work on PyPortal I had to add an additional exception type for `RuntimeError` to this except block. I'm guessing that ESP32-S2 might behave a bit differently than ESP32SPI in this regards. 

With the code like this:
```
except (RuntimeError, ConnectionError) as error:
```
I did get a successfull test on PyPortal Titano with 2 "bad" networks as the first two in my list and then a good network as the 3rd one. It does retry 10 times with 3 second pauses in-between each, so it took it a little bit to get to the good network. But once it got there it did successfully connect and start pulling data. 

My test was done with this project: https://learn.adafruit.com/pyportal-iss-tracker/overview

Once we add the RuntimeError catch here this change seems good to me though.

Thanks for tacking on this enhancement @tekktrik